-1 
Tested at: ec970ebb0df51044ecb4d41e994bb7722369e318
 I found an error when building: 
 
 <pre>/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-05-14-2300/src/FWCore/Framework/interface/stream/makeGlobal.h:48:27:   required from 'T* edm::stream::impl::makeStreamModule(const edm::ParameterSet&, const void*) [with T = ObjectSelector<CSCBeamHaloConfigSelector>]'
/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-05-14-2300/src/FWCore/Framework/interface/stream/ProducingModuleAdaptor.h:75:70:   required from 'edm::stream::ProducingModuleAdaptor<T, M, B>::setupStreamModules()::<lambda()> [with T = ObjectSelector<CSCBeamHaloConfigSelector>; M = edm::stream::EDFilterBase; B = edm::stream::EDFilterAdaptorBase]'
/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-05-14-2300/src/FWCore/Framework/interface/stream/ProducingModuleAdaptor.h:74:36:   required from 'struct edm::stream::ProducingModuleAdaptor<T, M, B>::setupStreamModules() [with T = ObjectSelector<CSCBeamHaloConfigSelector>; M = edm::stream::EDFilterBase; B = edm::stream::EDFilterAdaptorBase]::<lambda()>'
/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-05-14-2300/src/FWCore/Framework/interface/stream/ProducingModuleAdaptor.h:74:9:   required from 'void edm::stream::ProducingModuleAdaptor<T, M, B>::setupStreamModules() [with T = ObjectSelector<CSCBeamHaloConfigSelector>; M = edm::stream::EDFilterBase; B = edm::stream::EDFilterAdaptorBase]'
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-05-14-2300/src/CommonTools/RecoAlgos/interface/TrackSelector.h:135:9:   required from here
/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-05-14-2300/src/CommonTools/UtilAlgos/interface/ObjectSelector.h:54:5: error: no matching function for call to 'helper::NullPostProcessor<std::vector<reco::Track>, edm::EDFilter>::init(ObjectSelector<CSCBeamHaloConfigSelector>&)'
     postProcessor_.init(* this);
     ^
/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-05-14-2300/src/CommonTools/UtilAlgos/interface/ObjectSelector.h:54:5: note: candidate is:
In file included from /afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-05-14-2300/src/CommonTools/UtilAlgos/interface/ObjectSelector.h:24:0,
                 from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-05-14-2300/src/Alignment/CommonAlignmentProducer/plugins/AlignmentCSCBeamHaloSelectorModule.cc:4:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9081/4687/summary.html

--- a/CommonTools/RecoAlgos/interface/TrackSelector.h
+++ b/CommonTools/RecoAlgos/interface/TrackSelector.h
@@ -147,7 +147,7 @@ namespace helper {
 
 
   template<>
-  struct StoreManagerTrait<reco::TrackCollection> {
+  struct StoreManagerTrait< reco::TrackCollection, edm::stream::EDFilter<> > {
     typedef TrackCollectionStoreManager type;
     typedef TrackSelectorBase base;
   };

--- a/CommonTools/RecoAlgos/plugins/LargestEtCaloJetSelector.cc
+++ b/CommonTools/RecoAlgos/plugins/LargestEtCaloJetSelector.cc
@@ -6,12 +6,12 @@
  *
  */
 #include "FWCore/Framework/interface/MakerMacros.h"
-#include "CommonTools/UtilAlgos/interface/ObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/ObjectSelectorStream.h"
 #include "CommonTools/UtilAlgos/interface/SortCollectionSelector.h"
 #include "DataFormats/JetReco/interface/CaloJet.h"
 #include "CommonTools/Utils/interface/EtComparator.h"
 
-typedef ObjectSelector<
+typedef ObjectSelectorStream<
           SortCollectionSelector<
             reco::CaloJetCollection, 
               GreaterByEt<reco::CaloJet> 

--- a/CommonTools/RecoAlgos/plugins/LargestEtPFJetSelector.cc
+++ b/CommonTools/RecoAlgos/plugins/LargestEtPFJetSelector.cc
@@ -11,7 +11,7 @@
 #include "DataFormats/JetReco/interface/PFJet.h"
 #include "CommonTools/Utils/interface/EtComparator.h"
 
-typedef ObjectSelector<
+typedef ObjectSelectorStream<
           SortCollectionSelector<
             reco::PFJetCollection, 
               GreaterByEt<reco::PFJet> 

--- a/CommonTools/RecoAlgos/plugins/LargestEtPFJetSelector.cc
+++ b/CommonTools/RecoAlgos/plugins/LargestEtPFJetSelector.cc
@@ -6,7 +6,7 @@
  *
  */
 #include "FWCore/Framework/interface/MakerMacros.h"
-#include "CommonTools/UtilAlgos/interface/ObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/ObjectSelectorStream.h"
 #include "CommonTools/UtilAlgos/interface/SortCollectionSelector.h"
 #include "DataFormats/JetReco/interface/PFJet.h"
 #include "CommonTools/Utils/interface/EtComparator.h"

